src/miniscript/astelem.rs

```diff
@@ -80,7 +81,7 @@ impl<Pk: MiniscriptKey, Ctx: ScriptContext> Terminal<Pk, Ctx> {
         Pk::RawPkHash: 'a,
     {
         match *self {
-            Terminal::PkK(ref p) => pred(p),
+            Terminal::PkK(ref p) => p.for_each_key(pred),
```

Comment:
Add a test `for_each_key` for a fragment containing musig.

Reply:
Added a test for this `test_for_each_key` in `miniscript/mod.rs`